src/main/main.cpp

```diff
@@ -502,8 +557,20 @@ main(int argc, char* const* argv)
         case OPT_BASE64:
             base64 = true;
             break;
+        case OPT_CATCHUP_AT:
+            doCatchupAt = true;
+            catchupAtTarget = atol(optarg);
+            break;
         case OPT_CATCHUP_COMPLETE:
-            catchupComplete = true;
+            doCatchupComplete = true;
+            break;
+        case OPT_CATCHUP_RECENT:
+            doCatchupRecent = true;
+            catchupRecentCount = atol(optarg);
+            break;
+        case OPT_CATCHUP_TO:
+            doCatchupTo = true;
+            catchupToTarget = atol(optarg);
```

Comment:
given that this has the potential to blow up people's databases, we should not use `atol` to parse parameters. Also `0` should not be allowed as a magic number and people should use `current` or `latest` instead.

Reply:
what should we use instead? `atoi` or do you mean something else?

Reply:
you can use `std::stoi` (or other flavor like `std::stoul`), and make sure to use the optional `pos` parameter to ensure that the entire string is consumed.